compiler/optimizations/forallOptimizations.cpp

```diff
@@ -1140,6 +1140,7 @@ static Symbol *generateStaticCheckForAccess(CallExpr *access,
 
     VarSymbol *checkSym = new VarSymbol("chpl__staticAutoLocalCheckSym");
     checkSym->addFlag(FLAG_PARAM);
+    checkSym->addFlag(FLAG_TEMP);
```

Comment:
Adding this flag prevents normalization from adding PRIM_END_OF_STATEMENT in the wrong places for loops.